packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -0,0 +1,163 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ChangeFamily } from "../change-family";
+import { AnchorSet, Delta } from "../tree";
+
+export interface Commit<TChangeset> {
+    sessionId: SessionId;
+    seqNumber: SeqNumber;
+    refNumber: SeqNumber;
+    changeset: TChangeset;
+}
+
+type SessionId = number;
+type SeqNumber = number;
+
+/**
+ * Represents a local branch of a document and interprets the effect on the document of adding sequenced changes,
+ * which were based on a given session's branch, to the document history
+ */
+// TODO: Remove commits when they are no longer in the collab window
+// TODO: Try to reduce this to a single type parameter
+// TODO: Move logic into Rebaser if possible
+export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TChangeset>> {
```

Comment:
How does this relate to Rebaser?

I was expecting this to wrap a Rebaser. The values in the branches map would be RevisionTags for that rebaser, as would trunk (which I assume is the latest sequenced change?: could use more docs) and localChanges.

Reply:
I wanted to write this in a less abstract way to get a better understanding of what we want to be able to do before moving most of the logic into a module which isn't aware of the Fluid collaboration model. While it would probably be easy to do that refactoring now, I'd like to wait until we've done more design on how changesets will reference each other. It might be convenient and efficient for them to use sequence numbers, which will likely require more coupling with the EditManager.